challenge2.py

```diff
@@ -0,0 +1,82 @@
+"""
+    Advent of Code Day 2
+"""
+from itertools import groupby, combinations
+
+from common.input_file import read_strings
+
+
+def transform_input(box_id):
+    """
+        We sort and groupby to get lists of letters from each ID
+    """
+    grouped_box_id = groupby(sorted(box_id))
+
+    # We need to convert to a list so that we may go through this
+    # list more than once, otherwise we exhaust iterators first time through
+
+    return [(key, list(group)) for key, group in grouped_box_id]
```

Comment:
since all you need is the frequency here, could we just go ahead and use len(group)?

Reply:
That's a good improvement, thanks!

I'm going to keep it as is to reflect my original algorithm, but I think you're absolutely right.